### PR TITLE
php 8.1 : Fix deprecated

### PR DESCRIPTION
**Pull request by cboucaut**
Hi,

I use the projet with PHP 8.1 and I've found 2 deprecated.

### CODE REVIEW

File: src/Http/RequestMatcher/QuerystringRequestMatcher.php [hunk 1]
```diff
@@ -45,6 +45,6 @@ public function matches(Request $request): bool
             return true;
         }
 
-        return (bool) preg_match('{'.$this->queryString.'}', rawurldecode($request->getQueryString()));
+        return (bool) preg_match('{'.$this->queryString.'}', rawurldecode($request->getQueryString() ?? ''));
```
Comment (dbu) [inline src/Http/RequestMatcher/QuerystringRequestMatcher.php:48]: i think the $request must always exist and have the method, so this should be good enough:

```suggestion
        return (bool) preg_match('{'.$this->queryString.'}', rawurldecode($request->getQueryString() ?: ''));
```

**Review by dbu — COMMENTED** [review]
thanks for the fixes. one suggestion to be a bit more robust i think.

File: src/Http/RequestMatcher/QuerystringRequestMatcher.php (hunk 1 above)
Comment (cboucaut) [reply src/Http/RequestMatcher/QuerystringRequestMatcher.php]: Thank, i updated my PR with your suggestion :)

**Comment by dbu** [comment]
thanks a lot!

**Pull request merged**